lib/collection/src/shards/shard_holder/resharding.rs

```diff
@@ -513,6 +513,22 @@ impl ShardHolder {
         }
     }
 
+    pub async fn cleanup_local_shard(&self, shard_id: ShardId) -> CollectionResult<UpdateResult> {
+        let shard = self.get_shard(&shard_id).ok_or_else(|| {
+            CollectionError::not_found(format!("shard {shard_id} does not exist"))
+        })?;
+
+        if !shard.is_local().await {
+            return Err(CollectionError::bad_shard_selection(format!(
+                "shard {shard_id} is not a local shard"
+            )))?;
+        }
+
+        let filter = self.hash_ring_filter(shard_id).expect("hash ring filter");
+        let filter = Filter::new_must(Condition::CustomIdChecker(Arc::new(filter)));
+        shard.cleanup_local_shard(filter).await
```

Comment:
This appears to clean incorrectly. Based on testing I assume it may be deleting the inverse.

Here I used clean on shard 0. Total point count was 1000:

![image](https://github.com/user-attachments/assets/ab2330f6-d65d-46d8-bc8c-a5934edb315a)

Reply:
I patched this in https://github.com/qdrant/qdrant/pull/5024/commits/a008b47da9b2e35aafebd3a2e27d1a758a09be3c which seems to give the correct result.

1000 points over 3 shards after cleanup:
![image](https://github.com/user-attachments/assets/cc810ad1-459c-4433-bc04-b480b95c44aa)

Could you check whether using `must_not` is preferred versus inverting the condition in the `CustomIdChecker`?

Reply:
_I think_ you're right, this should be `must_not`. Thanks!